### ISSUE TITLE
Custom memory allocators

### ISSUE THREAD

**Issue by atticus5** [glouw/ctl#14]
Would like to also have the ability to define a custom memory allocator as we do for free and copy functions.

**Comment by glouw** [comment]
Yes, all memory allocations are defined with malloc, so just override malloc with your own call. Remember to revert the malloc redef at the end. This can be done on a per container basis, so it's pretty useful!

```
void* my_malloc(size_t size) { .... }
#define malloc my_malloc
#define P
#define T int
#include <vec.h>
#undef my_malloc
```

**Comment by graducan** [comment]
@glouw but what about the free function? I cannot override it, cause non-POD types use free as sufix..

**Comment by atticus5** [comment]
@glouw but can't do the same for free unfortunatelly, cause it is used as a suffix, unless I change it
